File: backend/src/PageNest.Domain/Entities/Book.cs
namespace PageNest.Domain.Entities;

public class Book
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Author { get; set; }
    public string Description { get; set; }
    public DateTime PublishedDate { get; set; }
    public string ISBN { get; set; }
    public int PageCount { get; set; }
    public string Language { get; set; }
    public string CoverImageUrl { get; set; }
    public int Stock { get; set; }
    public decimal Price { get; set; }
    public Guid CategoryId { get; set; }
    public Category Category { get; set; }
    public ICollection<BookGenre> BookGenres { get; set; }
    public ICollection<Review> Reviews { get; set; }
}

File: frontend/pagenest.client/src/types/book.ts
export interface IBook {
  id?: string;
  title: string;
  author: string;
  description: string;
  publishedDate: Date;
  isbn: string;
  pageCount: number;
  languageId: string;
  coverImageUrl: string;
  stock: number;
  price: number;
  categoryId: string;
}

export interface IBookRowProps {
  book: IBook;
  onEdit: () => void;
  onDelete: () => void;
  languageName: string;
  categoryName: string;
}

export interface IEditBookModalProps {
  isOpen: boolean;
  onClose: () => void;
  editedBook: Partial<IBook>;
  setEditedBook: (book: Partial<IBook>) => void;
  onSubmit: (e: React.FormEvent) => void;
}

export interface IDeleteBookModalProps {
  isOpen: boolean;
  onClose: () => void;
  onConfirm: () => void;
}

export interface IUseEditBookProps {
  books: IBook[];
  setBooks: React.Dispatch<React.SetStateAction<IBook[]>>;
  closeModal: () => void;
  setError: (error: string | null) => void;
}

export interface IUseDeleteBookProps {
  books: IBook[];
  setBooks: React.Dispatch<React.SetStateAction<IBook[]>>;
  closeModal: () => void;
  setError: (error: string | null) => void;
}
